lib/gadget/gadget.vala

```diff
@@ -304,6 +304,22 @@ namespace Frida.Gadget {
 				range: range
 			);
 		}
+
+#if ANDROID
+		construct {
+			if (executable_name.index_of ("app_process") >= 0) {
+				try {
+					string cmdline;
+					FileUtils.get_contents ("/proc/self/cmdline", out cmdline);
+					if (cmdline != "zygote" && cmdline != "zygote64") {
+						executable_name = cmdline;
+						cached_bundle_id = cmdline.substring(0, cmdline.last_index_of (":"));
```

Comment:
Missing space before opening parenthesis. However, I think this can be simplified by using Vala's string slicing syntax:

```suggestion
						cached_bundle_id = cmdline[0:cmdline.last_index_of (":")];
```

(Untested so would be good to verify that it supports negative indexing.)